### PR TITLE
build & test: dump data races to separate file

### PR DESCRIPTION
**Pull request by jmank88**

### CODE REVIEW

File: .github/workflows/pkg.yml [hunk 1]
```diff
@@ -18,7 +18,15 @@ jobs:
         run: go build -v ./...
 
       - name: Unit Tests
-        run: go test -race ./... -coverpkg=./... -coverprofile=pkg_coverage.out
+        run: GORACE="log_path=$PWD/race" go test -race ./... -coverpkg=./... -coverprofile=pkg_coverage.out
+
+      - name: Print Races
+        id: print-races
+        run: |
+          find race.* | xargs cat > race.txt
```
Comment (HenryNguyen5) [inline .github/workflows/pkg.yml:26]: more than one race file gets generated?

File: .github/workflows/pkg.yml [hunk 2]
```diff
@@ -18,7 +18,15 @@ jobs:
...
+          if [[ -s race.txt ]]; then
+            cat race.txt
+          fi
```
Comment (HenryNguyen5) [inline .github/workflows/pkg.yml:29]:
```suggestion
        run: cat race.*
```

Could we do this instead?

Comment (jmank88) [reply .github/workflows/pkg.yml:29]: Possibly, but I copied core: https://github.com/smartcontractkit/chainlink/blame/develop/.github/workflows/ci-core.yml#L182-L189
cc @momentmaker

Comment (momentmaker) [reply .github/workflows/pkg.yml:29]: I think the issue/bug before was when there is no `race.txt` that was output and this will fail

Comment (momentmaker) [reply .github/workflows/pkg.yml:29]: chatgpt says something like this?
```
  if ls race.* 1> /dev/null 2>&1; then
    cat race.* > race.txt
    cat race.txt
  fi
  ```

Comment (jmank88) [reply .github/workflows/pkg.yml:29]: Hmm is that an improvement?

Comment (momentmaker) [reply .github/workflows/pkg.yml:29]: good question - I think it's more a preference thing?

Comment (HenryNguyen5) [reply .github/workflows/pkg.yml:29]: Feel free to resolve this, if you copied it from core its fine as-is, i didnt realize we had an edge case to cover. If anything, a comment would be nice

**Pull request merged**